enhancements/installer/aws-internal-clusters.md

```diff
@@ -0,0 +1,192 @@
+
+#### Access to Internet
+
+The cluster still continues to require access to Internet.
+
+#### Resources created by the installer
+
+The installer will no longer create (vs. the fully-IPI flow with pre-existing VPC):
+
+- Public Load Balancers for API (6443) (aws_lb.api_external, aws_lb_target_group.api_external, aws_lb_listener.api_external_api)
+- No security group that allows 6443 from Internet.
+- Public DNS record (aws_route53_record.api_external)
+- No public IP address associated to bootstrap-host (associate_public_ip_address: false)
+- No security group that allows SSH to bootstrap-host from Internet (aws_security_group_rule.ssh)
+
+The installer will continue to create (vs. the fully-IPI flow with pre-existing VPC):
+
+- Private Load Balancers for API (6443) and MCS (22623)
+- DNS records in the private R53 Zone
+- Security Group that allows SSH to the bootstrap-host from the inside the VPC.
+
+#### Bootstrap instance placement
+
+The Bootstrap instance is placed in one of the provided private subnet, compared to the External case it's placed in one of the public subnet.
+
+#### Operators
+
+##### Ingress
+
+The `default` ingresscontroller for the cluster on AWS has LoadBalancer scope set to External as per API [reference][ingresscontroller-default]. But for `Internal` clusters the `default` ingresscontroller for the cluster is explicity set to Loadbalancer scope `Internal`.
+
+#### Limitations
+
+- The Kubernetes API endpoints cannot be made public after installation using some configuration in the cluster. The users will have to manually choose the public subnets from the VPC where the cluster is deployed, created public load balancer with control-plane instances as backend and also ensure the control-plane security groups allow traffic from Internet on 6443 (Kubernetes API port). Also the user will have to make sure they pick public subnets in each Availability Zone as detailed in [documentation][public-lb-to-private-instances].
+
+### Risks and Mitigations
+
+#### Public Service type Load Balancers
+
+Creating Service type Loadbalancer that are public would require users to select public subnets and tag them `kubernetes.io/cluster/<cluster-infra-id>: shared` so that cloud provider can use them to create public load balancers. The choice of public subnets is similarly strict as mentioned [limitation][#limitations].
+
+#### Public OpenShift Ingress
+
+Changing the OpenShift Ingress to be public after installation requires extra steps for users as detailed in previous [section][#public-service-type-load-balancers]
+
+## Design Details
+
+### Test Plan
+
+The containers in the CI cluster that run the installer and the e2e tests will create VPN connection to public VPN endpoint created in the network deployment in the CI AWS account. This will provide the client transparent access to otherwise internal endpoints of the newly created cluster.
+
+### Graduation Criteria
+
+*This enhancement will follow standard graduation criteria.
```

Comment:
nit: trailing `*` or drop the leading one

Reply:
done https://github.com/openshift/enhancements/compare/d1df0a7526e3ead64c786034f1f50e29d556a5aa..d5621ab1053e0b04209359ebcd82d867ab38ac2c